src/TLuaInterpreter.cpp

```diff
@@ -418,62 +418,80 @@ int TLuaInterpreter::resetProfile( lua_State * L )
+// same name
+// selectString( [windowName], text, number_of_match )
+// Will now consider an EMPTY window name or the literal "main" as being the
+// same as an omitted windowName - i.e. is the main console window.
+int TLuaInterpreter::selectString( lua_State * L )
 {
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if( ! pHost ) {
+        lua_pushstring( L, tr( "selectString: NULL Host pointer - something is wrong!" )
+                        .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+
     int s = 1;
-    int n = lua_gettop( L );
-    string a1;
-    if( n > 2 )
-    {
-        if( ! lua_isstring( L, s ) )
-        {
-            lua_pushstring( L, "select: wrong argument type" );
-          lua_error( L );
-          return 1;
+    QString windowName; // only for 3 argument case, will be null if not assigned to which is different from being empty
+    if( lua_gettop( L ) > 2 ) {
+        if( ! lua_isstring( L, s ) ) {
+            lua_pushstring( L, tr( "selectString: bad argument %1 type (window name as string, is optional {defaults"
+                                   "to \"main\" if omitted}, got %2!)" )
+                            .arg( s )
+                            .arg( luaL_typename( L, s ) )
+                            .toUtf8().constData() );
+            lua_error( L );
+            return 1;
         }
-        else
-        {
-            a1 = lua_tostring( L, s );
+        else {
+            // We cannot yet properly handle non-ASCII windows names but we will eventually!
+            windowName = QString::fromUtf8( lua_tostring( L, s ) );
+            if( ! windowName.compare( tr( "main" ), Qt::CaseSensitive ) ) {
+                // This matches the identifier for the main window - so make it
+                // appear so by emptying it...
+                windowName = QString();
+            }
             s++;
         }
     }
-    string luaSendText="";
-    if( ! lua_isstring( L, s ) )
-    {
-        lua_pushstring( L, "select: wrong argument type" );
+
+    QString searchText;
+    if( ! lua_isstring( L, s ) ) {
+        lua_pushstring( L, tr( "selectString: bad argument %1 type (text to search for as string expected, got %2!)" )
+                        .arg( s )
+                        .arg( luaL_typename( L, s ) )
+                        .toUtf8().constData() );
         lua_error( L );
         return 1;
     }
-    else
-    {
-        luaSendText = lua_tostring( L, s );
+    else {
+        searchText = QString::fromUtf8( lua_tostring( L, s ) );
+        // CHECK: Do we need to qualify this for a non-blank string?
         s++;
     }
-    int luaNumOfMatch;
-    if( ! lua_isnumber( L, s ) )
-    {
-        lua_pushstring( L, "select: wrong argument type" );
+
+    qint64 numOfMatch = 0;
+    if( ! lua_isnumber( L, s ) ) {
+        lua_pushstring( L, tr( "selectString: bad argument %1 type (match number as number expected, got %2!)" )
```

Comment:
Ah, right, reading the Wiki reveals that it is a specific instance to select that it means...

:beetle:  Ohh, that is another mistake - I forget the `#` we use everywhere when we are talking about arguments in those help messages - there are 3 instances in this function that need a fixup! :hammer_and_wrench: